Add data attribute to multirun

ash2k opened the pull request:
Fixes #78.

@rockwotj could you try the version from this branch to see if it works for you?

rockwotj commented:
@ash2k thanks! This does not seem to work for me.

Here is the test I ran:

```python
genrule(
    name = "data_file",
    srcs = [],
    outs = ["file.dat"],
    cmd = "echo 'foo' > \"$@\"",
)

command(
    name = "cmd",
    arguments = [
        "--data_file=$(location :data_file)",
    ],
    command = "//:binary_rule",
    data = [":data_file"],
)
```

If you look in `bazel-bin/cmd.bash.runfiles/__main__/` file.dat is not there.

also the args stuff is what I mean by #79, sorry that wasn't clear before :/

rockwotj commented on multirun/def.bzl:
```diff
@@ -25,6 +25,11 @@ def _multirun_impl(ctx):
     runfiles = runfiles.merge(runnerInfo.default_runfiles)
 
+    for data_dep in ctx.attr.data:
+        default_runfiles = data_dep[DefaultInfo].default_runfiles
```
I have not actually written any rules, but this seems to be getting the dep's runfiles, when don't you want the dep *as a runfile*? I think this will get the transitive deps but not the direct dep.

rockwotj replied:
Maybe the right fix is something like:

```python
for data_dep in ctx.attr.data:
  runfiles = runfiles.merge(ctx.runfiles(files = data_dep))
```

ash2k replied:
Yeah, this is what was not clear to me, the docs are vague. I'll update the PR.

ash2k commented:
@rockwotj I have rebased the branch to incorporate the bugfix from #81 and also updated the code to not introduce the same mistake. I have not tested it but I'll add tests later, perhaps tomorrow. Ran out of time today. You may give it a try if you want.
